app/src/main/java/net/gsantner/markor/format/markdown/MarkdownTextModuleActions.java

```diff
@@ -489,4 +509,190 @@ public void onClick(DialogInterface dialog, int id) {
+        timePickButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View v) {
+                new TimePickerDialog(getActivity(),
+                        new TimePickerDialog.OnTimeSetListener() {
+                            @Override
+                            public void onTimeSet(TimePicker timePicker, int hour, int min) {
+                                datetime.set(Calendar.HOUR_OF_DAY, hour);
+                                datetime.set(Calendar.MINUTE, min);
+                                datetimeTextView.setText(parseDateTimeToCustomFromat(
+                                        timeFormatEditText.getText().toString(), datetime.getTimeInMillis()));
+                            }
+                        }, datetime.get(Calendar.HOUR_OF_DAY), datetime.get(Calendar.MINUTE),
+                            true).show();
+            }
+        });
+
+        // set builder and implement buttons to discard and submit
+        builder.setView(view)
+                .setNegativeButton(android.R.string.cancel, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialogInterface, int i) {
+
+                    }
+                })
+                .setPositiveButton(android.R.string.ok, new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int id) {
+                        _hlEditor.insertOrReplaceTextOnCursor(getOutput(
+                                selectFormatCheckBox.isChecked(),
+                                datetimeTextView.getText().toString(),
+                                timeFormatEditText.getText().toString()));
+                    }
+                });
+
+        builder.show();
+    }
+
+
+    /**
+     *
+     * @param timeFormat
+     *          {@link String} text which 'll be used as format for {@link SimpleDateFormat}
+     * @param datetime
+     *          {@link Long} selected datetime in milisecond
+     * @return
+     *          formated datetime
+     */
+    private String parseDateTimeToCustomFromat(String timeFormat, Long datetime) {
+        try {
+            DateFormat formatter = new SimpleDateFormat(timeFormat);
+            return formatter.format(datetime);
+        } catch (Exception e) {
+            // maybe we can add there some exception handler
+            return null;
+        }
+    }
+
+    /**
+     *
+     * @param isDatetimeSelected
+     *          {@link Boolean} information if we want datetime or format
+     * @param datetime
+     *          selected datetime as {@link String} based on given format
+     * @param format
+     *          {@link String} pattern used to convert datetime into text output
+     * @return
+     *          @datetime or @format, based on @isDatetimeSelected
+     */
+    private String getOutput(Boolean isDatetimeSelected, String datetime, String format) {
```

Comment:
Please use braces, even if not required. 
You can also use ternary operator if you want (` return condition ? datetime : format`), but no must.